### ISSUE TITLE
OsgiTestBundleProject should override readArtifactKey

### ISSUE THREAD

**Issue by lppedd** [eclipse-tycho/tycho#3690]
Playing with `tycho-baseline-plugin` I've noticed that 
```java
projectManager.getArtifactKey(project);
```
would return a key with a type of `TYPE_ECLIPSE_PLUGIN` even for `eclipse-test-plugin` projects.  
This means baseline checks are performed for test bundles, which in my opinion is inconvenient (if not wrong).

This happens because `OsgiTestBundleProject` does not override `readArtifactKey`, which always returns
```java
new DefaultArtifactKey(ArtifactType.TYPE_ECLIPSE_PLUGIN, getBundleSymbolicName(), getBundleVersion());
```
through `OsgiManifest`.

A good solution might be to override that method
```java
@Component(role = TychoProject.class, hint = PackagingType.TYPE_ECLIPSE_TEST_PLUGIN)
public class OsgiTestBundleProject extends OsgiBundleProject {
    @Override
    public ArtifactKey readArtifactKey(final File location) {
        final OsgiManifest mf = bundleReader.loadManifest(location);
        return mf.toArtifactKey(ArtifactType.TYPE_ECLIPSE_TEST_PLUGIN);
    }
}
```
allowing passing in an explicit type to `OsgiManifest.toArtifactKey`.

**Comment by laeubi** [comment]
> would return a key with a type of TYPE_ECLIPSE_PLUGIN even for eclipse-test-plugin projects.

That is correct, there is no difference between the both beside the packaging.

> This means baseline checks are performed for test bundles, which in my opinion is inconvenient (if not wrong).

Depends on the use-case, one usually want to disable the check for example by specify a property for such projects that then skips the execution:

https://tycho.eclipseprojects.io/doc/latest/tycho-baseline-plugin/verify-mojo.html#skip

**Comment by lppedd** [comment]
@laeubi skipping is what I do now, but imaging having 100+ test bundles, and having to remember to skip every time.

Not the best DevEx.  
IMO the best approach would be enabling test-bundles baseline checks with a single parameter in the root configuration.

**Comment by laeubi** [comment]
> skipping is what I do now, but imagine having 100+ test bundles, and having to remember to skip every time.

There is no such thing as a "test-bundle" and that's why eclipse-test-plugin is actually discouraged, so the general rule is to not build any new features on this bad concept.

it was just used as a vehicle for a different workflow that is not required anymore and as "test bundles" almost never export any API checking them should be not an issue at all anyways.

> IMO the best approach would be enabling test-bundles baseline checks with a single parameter in the root configuration.

If you have a classical pom layout you would do that in the parent `pom.xml`, for [structured layouts / pomless](https://tycho.eclipseprojects.io/doc/latest/StructuredBuild.html) you can create a `build.properties` file in your `tests` folder and specify

```
pom.model.property.tycho.baseline.skip
```

**Comment by lppedd** [comment]
I don't have a pom-less build, but I can reproduce a similar layout and have a single point where I specify the skipping property.

I think I can close this for now.

**Comment by laeubi** [comment]
It works of course also in a (child parent) pom.xml :-)
